### DISCUSSION TITLE
배하늘 스토리

### DISCUSSION THREAD

**Discussion by haneeeul** [ondal-house/zu-story#18] [스토리 작성 안건]
# 주류별 리뷰가 가능한 온라인 검색&커뮤니티 사이트 만들기
서비스명(가칭): 기억나니
서비스 목표: 술 리뷰를 바탕으로 하는 검색&커뮤니티 만들기
타켓 유저:  
> 데이터를 수집하기 좋은 특별 유저층
  1. 술에 대해 잘 알고, 자신이 아는 술 지식을 나누고 싶은 사람
> 서비스 인지도 향상을 위한 유저층
  2. 술에 대해 잘 모르고, 자신에게 맞는 술을 알고 싶은 사람(서비스 인지도 향상을 위한 유저층)
  3. (구글, 네이버 등 기존의 검색엔진 검색결과가 너무 방대하다고 느껴서)원하는 술을 찾기 위해 술만 검색하고 싶은 사람
  
domain: doyouremember.com

## 기능 / 정책 기획
- 비회원/회원 리뷰 열람 및 리뷰의 '좋아요' 버튼 누르기 가능
  - **추후에 스페셜 리뷰(특정 회원만 열람 가능) 기능을 만들지는 합의 필요**
- 비회원 리뷰 작성은 불가능하며 회원가입은 소셜만 허용
  - 연령 제한을 일반 ID/PW 로그인으로는 하기 쉽지 않을 것 같아서 해당 정책 사용
  - **몇가지 소셜 회원가입 및 로그인을 허용할지는 합의 필요**
- 리뷰 작성 시 이미지는 최대 10장까지 업로드 가능하며 한번에 업로드하는 이미지의 총 용량은 00MB 를 넘지 않음
  - 용량 초과 시 화면에서 업로드 불가 메시지 출력
  - **00MB는 CDS와 협의 필요**
- 리뷰 작성 시 텍스트는 한글 최대 000자, 영어 최대 000자 제한
- 각 리뷰는 작성자가 수정 가능
  - 수정 시 리뷰에 수정일시, '(수정됨)' 추가로 표시

### 각 데이터별 필요 요소 정리
-  주류 설명
제품명/제조사/가격(어디 기준?)/도수/유형(발효주,증류주)
- 각 주류에 대한 리뷰
작성자 닉네임/리뷰 내용/리뷰 이미지(n장 이하)/작성일시/수정일시(+ '수정됨' 표시 화면에 추가)/리뷰 추천or비추천 버튼

## 화면 기획
- 반응형 웹페이지로 제작
  - **mobile first?**
  - **mobile first 면 차라리 앱은 어떤지?**
- **와이어 프레임 작업 필요한가?**
- **피그마 (상세 UI 디자인을 위한 작업도구) 만들어야 할까?**

### 화면 상세 기획: main page
URL : /
주류 카테고리
- 화면에 대분류 5개만 노출
- '더보기' 클릭 시 검색 및 카테고리 페이지로 이동
카테고리 별 추천 제품 캐로셀
- 최대 10개 제품 노출
  - 이미지, 제품명, '더보기' 노출
(가칭)좋아요를 많이 받은 리뷰
- 최대 5개 리뷰 노출
  - 리뷰 작성자 닉네임, 텍스트 리뷰 2줄 화면에 노출 후 ... 처리, '더보기' 버튼
푸터
  - 서비스명, 카피라이터, 우리 깃헙 주소 및 이메일 주소 노출
 
### 화면 상세 기획: 주류 카테고리 page
URL : /category
**사실 가장 중요한 카테고리의 종류는... 감이 안오는군요**
- 사용자가 메인 페이지에서 '카페고리 더보기'를 눌러 카테고리 페이지 진입시 너무 많은 카테고리에 압도되지 않도록 종류는 20가지 안쪽으로만 선정
- 카테고리 페이지 진입 시 페이지 상단에 검색창 노출
  - 제품명/제조사/유형/카테고리 명으로 text 검색 가능(like 검색 가능인지는 생각해보자)
  - 제조사/유형/카테고리/가격대/도수 기준으로 필터링 가능

### 화면 상세 기획: 카테고리 별 추천 제품 목록 및 리뷰 목록 page
URL : /liquor  or /review or /list (**맞나요?**)
- 메인 페이지에서 카테고리 별 추천 제품 캐로셀 클릭 시 추천 제품 페이지 진입
- 추천의 우선순위는 '리뷰 개수'(개수가 많으면 추천 페이지에서 상위권에 노출됨)로만 판단
  - **리뷰의 '추천(좋아요 표시)' 개수가 해당 리뷰에 영향을 줄지는 합의 필요**
  - **이유: 비로그인 회원도 누를 수 있어서**  
  

  - 가격대별 추천
  - 카테고리별 추천
  - 추천 리뷰(추천을 많이 받은 리뷰 순서대로 노출)

#### FE
typescript
framework : react, bootstrap 또는 우리들의 친구 Material UI ^_^ 좀 벗어나보고 싶은데 어렵다 semantic UI 는 또 뭔지.. tailwind / Antd
style(CSS) : sass
- Test Framework
Jest, Cypress ... (써본게 너무 제한적이라 도입할만한게 뭔지 모르겠음)

#### BE
java + spring boot + spring security  
API 문서 : Spring Rest Docs + Git Wiki
> 너 Spring Rest Docs 가 뭔지 모르니...? Git Wiki 라니 무슨 말이지?
>  - Spring Rest Docs 만 사용하기에는 부적절하다고 생각함(서버내리면 볼 수가 없음)
>  - 한달에 한 번, 날잡고 Spring Rest Docs - Git Wiki 간 sync 작업 하면 될 것 같음

or

typescript + node.js
API 문서 : APIDocs + Git Wiki
_(미안하다.. 여기는 ㄹㅇ 맨땅에 헤딩이다.. 그래도 제안해본다..)_

#### Database
Mysql
  - 외에 우리가 선택할 수 있는 관계형 데이터 베이스 있는지... 모르겠음

#### CI/CD
@gychoi0916 @ResinCRT 도와줘요~!

**Comment by haneeeul** [comment]
1. 역할 분담을 어떻게 가져갈 것인지
2. ERD 언제 그리는지(애초에 한번은 그리고 가야할 것 같음)
3. ground rule 정리가 필요해 보인다.

**Reply by ypd01018** [reply]
1번
희망 역할 포지션이 몰린다 -> 풀스택
의외로 균등하다 -> 해당 스택 선택 및 나누기 -> 이번 회의에서 결정하면 될 것 같음
이면 좋을 것 같아요.

2번
ERD가 핵심 기능 선정이 완료되었다면 그리면 될 것 같아요

3번
좋아용
어떤 목표의 그라운드 룰?

**Comment by ypd01018** [comment]
### mobile first면 앱 어떨지
- 좋아요. 
- 그러나 웹을 포기해야한다는 점에서 접근성이 떨어질 수 있지 않을까? 검색엔진을 못탈텐데 홍보에 어렵지 않을까? 가 좀 걱정되어요.
- 그리고 각 개발자들이 PC가 모바일 개발의 부하를 견딜 수 있을지가 조금 무섭네요... 특히 윈도우 PC쓰시는 분들은 IOS가...
- IOS / Android 각 네이티브 개발은 러닝 커브가 높고 어려울 것 같아요. 또한 2개를 만들자니 빡세고, 하나만 하자니 아쉽고.. 플러터 / RN ㄱ? 이건 이거대로 네이티브 기능 개발 필요할 때 이슈가 있긴하지만...라이브러리 믿고 ㄱ?

### 와이어 프레임과 피그마
- 여기 어려움과 시간이 꽤 들 것 같은데 그렇다고 안하자니 서로 생각한 바가 다른 것을 말로만 전달하면 분명 다른 부분이 있을 것 같아서 아쉽네요...
- 어카지...

### sass
- scss 가 전 좀 더 명시적이고 좋은 것 같아요 ㅋㅋ

### css 프레임 워크
- 도입하자니 뭔가 저런 어드민페이지 / 백오피스 같은 UI는 좀....
- 근데 안하자니 퍼블리싱 너무 힘들겟다 생각이...

### Spring Rest Docs
- 이거 들어만 봤지 알아본적이 없는데 Swagger VS Rest Docs로 해서 장단점 잘 비교해서 선택해보면 좋을 것 같네요
- 서버 내리면 볼 수 없다하지만 Dev 서버가 따로 떠있을수도 있을 것같은데 이럴 경우는 wiki에 작성이 불필요해보이는데 그럼에도 선택 이유가 있을까요?

### Spring Security
- 이거 저도 고민해봤는데 도입은 좀 고민해봐야할 것 같아요 ㅋㅋ
- 스프링 시큐리티가 러닝커브가 좀 있는 편이라, 인터셉터에서 필요한 작업을 해서 로직 / 다른 걸 좀 더 집중할지, 이번 기회에 스프링 시큐리티를 도입해서 배워볼지 잘 선택해봐야할 것 같네요.

### DB
- MySQL이면 족할 것 같아요. MariaDB도 있긴한데 뭔가 선택 이유에 대해 마땅한 이유를 잘 모르겠네요 ㅋㅋㅋ
